src/deprecation.h

```diff
@@ -2,18 +2,22 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or https://www.opensource.org/licenses/mit-license.php .
 
+#include "consensus/params.h"
```

Comment:
It seems that in all header files including a header in our project the `#ifndef ...` is before the includes:

https://github.com/zcash/zcash/blob/master/src/key.h#L7-L10
https://github.com/zcash/zcash/blob/master/src/hash.h#L6-L9
https://github.com/zcash/zcash/blob/master/src/consensus/upgrades.h#L5-L8
...

I think we better change to that in here.